e2e/orchestrator_test.go

```diff
@@ -0,0 +1,303 @@
+package e2e
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/celestiaorg/orchestrator-relayer/cmd/qgb/helpers"
+	qgbtesting "github.com/celestiaorg/orchestrator-relayer/testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestOrchestratorWithOneValidator(t *testing.T) {
+	if os.Getenv("QGB_INTEGRATION_TEST") != TRUE {
+		t.Skip("Skipping QGB integration tests")
+	}
+
+	network, err := NewQGBNetwork()
+	HandleNetworkError(t, network, err, false)
+
+	// to release resources after tests
+	defer network.DeleteAll() //nolint:errcheck
+
+	// start 1 validator
+	err = network.StartBase()
+	HandleNetworkError(t, network, err, false)
+
+	// add orchestrator
+	err = network.Start(Core0Orch)
+	HandleNetworkError(t, network, err, false)
+
+	ctx := context.Background()
+	err = network.WaitForBlock(ctx, int64(network.DataCommitmentWindow+50))
+	HandleNetworkError(t, network, err, false)
+
+	// create dht for querying
+	bootstrapper, err := helpers.ParseAddrInfos(network.Logger, BOOTSTRAPPERS)
+	HandleNetworkError(t, network, err, false)
+	_, _, dht := qgbtesting.NewTestDHT(ctx, bootstrapper)
+	defer dht.Close()
+
+	err = network.WaitForOrchestratorToStart(ctx, dht, CORE0EVMADDRESS)
+	HandleNetworkError(t, network, err, false)
+
+	// give the orchestrators some time to catchup
+	time.Sleep(30 * time.Second)
+
+	vsConfirm, err := network.GetValsetConfirm(ctx, dht, 1, CORE0EVMADDRESS)
+	// assert the confirm exist
+	assert.NoError(t, err)
+	require.NotNil(t, vsConfirm)
+	// assert that it carries the right evm address
+	assert.Equal(t, CORE0EVMADDRESS, vsConfirm.EthAddress)
+
+	dcConfirm, err := network.GetDataCommitmentConfirmByHeight(ctx, dht, network.DataCommitmentWindow-2, CORE0EVMADDRESS)
+	// assert the confirm exist
+	require.NoError(t, err)
+	require.NotNil(t, dcConfirm)
+	// assert that it carries the right evm address
+	assert.Equal(t, CORE0EVMADDRESS, dcConfirm.EthAddress)
+}
+
+func TestOrchestratorWithTwoValidators(t *testing.T) {
```

Comment:
a quick doc to explain what is being tested here would be nice, but I'm not blocking on it

Reply:
https://github.com/celestiaorg/orchestrator-relayer/issues/205